blocks/pipeline.py

```diff
@@ -52,11 +52,13 @@ def fit(self, store: Store, plugins: List["Plugin"]) -> None:
         """Begin"""
         last_output = process_block(self.datasource, store)
         for plugin in plugins:
-            store, last_output = plugin.on_fit_begin(store, last_output)
+            store, last_output, trainer_callbacks = plugin.on_fit_begin(
```

Comment:
is this a generalizable pattern? can others models ever deal with this specific kind of function, or is it huggingface only? if it is huggingface only, can we somehow inject it into the HuggingfaceConfig instead?